src/helpers/pivot/spreadsheet_pivot/spreadsheet_pivot.ts

```diff
@@ -0,0 +1,384 @@
+import { InitPivotParams, Pivot } from "../../../types/pivot_runtime";
+import { toXC } from "../../coordinates";
+import { MONTHS, isDateTimeFormat } from "../../format";
+import { AGGREGATORS_FN, MEASURES_TYPES, toDomainArgs } from "../pivot_helpers";
+import { PivotParams } from "../pivot_registry";
+import {
+  DataEntries,
+  DataEntry,
+  dataEntriesToSpreadsheetPivotTable,
+} from "./data_entry_spreadsheet_pivot";
+import { createDate } from "./date_spreadsheet_pivot";
+import { SpreadsheetPivotRuntimeDefinition } from "./runtime_definition_spreadsheet_pivot";
+import { SpreadsheetPivotTable } from "./table_spreadsheet_pivot";
+
+interface SpreadsheetPivotParams extends PivotParams {
+  definition: SpreadsheetPivotCoreDefinition;
+}
+
+/**
+ * This class represents a pivot table that is created from a range of cells.
+ * It will extract the fields from the first row of the range and the data from
+ * the rest of the rows.
+ */
+export class SpreadsheetPivot implements Pivot<SpreadsheetPivotRuntimeDefinition> {
+  readonly type = "SPREADSHEET";
+  private getters: Getters;
+  private _definition: SpreadsheetPivotRuntimeDefinition | undefined;
+  private coreDefinition: SpreadsheetPivotCoreDefinition;
+  /**
+   * This array contains the data entries of the pivot. Each entry is an object
+   * that contains the values of the fields for a row.
+   */
+  private dataEntries: DataEntries = [];
+  private fields: PivotFields = {};
+  /**
+   * This array contains the keys of the fields. It is used to keep the order
+   * of the fields as they are in the range.
+   */
+  private fieldKeys: TechnicalName[] = [];
+  /**
+   * This object contains the pivot table structure. It is created from the
+   * data entries and the pivot definition.
+   */
+  private table: SpreadsheetPivotTable | undefined;
+  /**
+   * This error is set when the range is invalid. It is used to show an error
+   * message to the user.
+   */
+  private invalidRangeError: EvaluationError | undefined;
+  /**
+   * This flag is used to know when the pivot needs to be reloaded. It's only
+   * used in the evaluation process. At the end of each cycle, the flag is set
+   * to true so the pivot is reloaded in the next cycle.
+   */
+  isDirtyForEvaluation: boolean = true;
+
+  constructor(custom: ModelConfig["custom"], params: SpreadsheetPivotParams) {
+    this.getters = params.getters;
+    this.coreDefinition = params.definition;
+  }
+
+  init(params: InitPivotParams = {}) {
+    if (!this._definition || params.reload) {
+      this.invalidRangeError = undefined;
+      if (this.coreDefinition.dataSet) {
+        const { zone, sheetId } = this.coreDefinition.dataSet;
+        const range = this.getters.getRangeFromZone(sheetId, zone);
+        try {
+          ({ fields: this.fields, fieldKeys: this.fieldKeys } = this.extractFieldsFromRange(range));
+        } catch (e) {
+          this.fields = {};
+          this.fieldKeys = [];
+          this.invalidRangeError = e;
```

Comment:
This is a lie. `e` is implicitly `any` and could be different from `EvaluationError` (and it is `Error` in many cases)

Reply:
`private invalidRangeError: EvaluationError | Error | undefined;`
I think it's the right way to fix, what do you think ?